notebook related documentation

orifox opened the pull request:
New documentation specific to working in a Notebook.  I wanted to pull out a lot of information that is currently buried or non-existent into one place.  That means there is a little redundancy.  I don't mind that for the time being.  If these changes look correct, we can worry about how to consolidate all the information and potentially reorganize in the future.

EDIT: Rendered doc at https://jdaviz--832.org.readthedocs.build/en/832/index.html

EDIT: Fix #839

pllim commented:
You forgot to include your new page in root level index, please fix. Thanks!

`docs/notebook/index.rst: WARNING: document isn't included in any toctree`

pllim commented:
Update: I pushed a commit to fix the doc build.

pllim commented:
For future reference -- FYI:

* You can try to build the doc locally first by running `pip install -e . ; cd docs; make html` to make sure there are no errors/warnings.
* If you want to use a new section reference, you have to create the new tags too. They do not magically create themselves, unlike some other software you might be used to.

pllim commented on docs/notebook/export_data.rst:
```diff
@@ -0,0 +1,104 @@
+
+In `Imviz`, you can extract positional regions::
+
+    regions=myinstance.get_interactive_regions()
```
This comment applies to all similar occurences. Should the example more closely match the actual example notebooks?

```suggestion
    regions = viz.get_interactive_regions()
```

pllim commented on docs/notebook/import_data.rst:
```diff
@@ -0,0 +1,44 @@
+    from jdaviz.configs.mosviz.helper import MosViz as Mosviz
+    myinstance = Mosviz()
+    myinstance .app
+    import tempfile
+    data_dir = tempfile.gettempdir()
+    mosviz.load_data(directory=data_dir, instrument="nirspec") # Or niriss
```
This is all fake code anyway, no reason to introduce "tempfile" here to complicate things.

```suggestion
    mosviz.load_data(directory="/path/to/data/file", instrument="nirspec") # Or niriss
```

pllim reviewed (commented):
Thanks for the documentation! Definitely good to have something. We can polish things up to the standards of Astropy Tutorials in the future.

All the intersphinx stuff is still broken but I guess we can defer that. Seems to be a problem across the board.

Minor comments below.

kaya4me2 commented:
Documentation link still not quite correct ... in order to read in ascii file an initiate the Jdaviz interface had to to the following in the Jupiter notebook code: (i.e., the myinstance.app line placement )

<img width="1134" alt="correct_nujdaviz_ascii_call" src="https://user-images.githubusercontent.com/70973871/132402607-efbc50e2-8049-4a57-8f18-fc03a4589445.png">

Note that the flux density axis has no units on the axis, but the spectra looks about right when plotted in the visualizer.

Secondly, why aren't the uncertainties plotted ?

javerbukh commented:
@kaya4me2 Thank you for your comment! Would you be willing to provide the file you are trying to load for testing purposes?

When you mention the flux density axis, are you referring to the data itself or the plot?

Uncertainties can be plotted by running the follow: `myinstance.app.get_viewer("spectrum-viewer").show_uncertainties()`. Would you prefer uncertainties were plotted by default?

orifox commented:
Addressed all comments above, except:
1) loading uncertainties
2) Y-axis labeling problems (I noticed this myself with different data sets)  

Created new tickets for both.

This PR is now ready for final review.  Already labeled.

javerbukh commented on docs/notebook/export_data.rst:
```diff
@@ -0,0 +1,104 @@
+
+In `Imviz`, you can extract positional regions::
+
+    regions=myviz.get_interactive_regions()
```
``` suggestion
    regions = myviz.get_interactive_regions()
```
Minor nitpick.

javerbukh commented:
@orifox One minor comment, otherwise looks good! Ping me when you commit that change and I'll approve and merge.

kaya4me2 commented:
> @kaya4me2 Thank you for your comment! Would you be willing to provide the file you are trying to load for testing purposes?
> 
> When you mention the flux density axis, are you referring to the data itself or the plot?
> 
> Uncertainties can be plotted by running the follow: `myinstance.app.get_viewer("spectrum-viewer").show_uncertainties()`. Would you prefer uncertainties were plotted by default?

Yes that would be preferred (sorry about the delay in response), also where does the suggest line
myinstance.app.get_viewer("spectrum-viewer").show_uncertainties() come the flow of initiating the spexviz app in the notebook?

javerbukh commented:
It would come after data has been loaded into the application.

javerbukh reviewed (approved):
Looks good!
